File: tests/fixtures/go_sample/greeter.go
// Tiny Go fixture for the v0.2 tags + graph layer.
package sample

import "fmt"

type Greeter struct {
	name string
}

type Named interface {
	Name() string
}

func (g *Greeter) Name() string {
	return g.name
}

func (g *Greeter) Greet() string {
	return formatMessage(g.name)
}

func formatMessage(name string) string {
	return fmt.Sprintf("hello, %s", name)
}

File: tests/fixtures/java_sample/Greeter.java
// Tiny Java fixture for the v0.2 tags + graph layer.
package sample;

public class Greeter implements Named {
    private final String name;

    public Greeter(String name) {
        this.name = name;
    }

    @Override
    public String name() {
        return name;
    }

    public String greet() {
        return formatMessage(name);
    }

    private static String formatMessage(String n) {
        return "hello, " + n;
    }
}

interface Named {
    String name();
}

enum Severity {
    LOW,
    HIGH
}
